fmralign/alignment_methods.py

```diff
@@ -731,3 +739,255 @@ def transform(self, X, verbose=False):
+            print("Geometry embedding computed")
+
+        source_features_normalized = self._normalize(X.T)
+        target_features_normalized = self._normalize(Y.T)
+
+        if verbose:
+            print("Features computed")
+
+        if method == "dense":
+            mapping = FUGW(
+                alpha=self.alpha_coarse,
+                rho=self.rho_coarse,
+                eps=self.eps_coarse,
+                reg_mode=self.reg_mode,
+                divergence=self.divergence,
+            )
+
+            mapping.fit(
+                source_features=source_features_normalized,
+                target_features=target_features_normalized,
+                source_geometry=geometry_embedding_normalized
+                @ geometry_embedding_normalized.T,
+                target_geometry=geometry_embedding_normalized
+                @ geometry_embedding_normalized.T,
+                verbose=verbose,
+                **kwargs,
+            )
+
+            self.mapping = mapping
+
+        elif method == "coarse-to-fine":
+            # Subsample vertices as uniformly as possible on the surface
+            sampled_geometry = self._sample_geometry(
+                segmentation, geometry_embedding, n_samples
+            )
+
+            if verbose:
+                print("Samples computed")
+
+            coarse_mapping = FUGW(
+                alpha=self.alpha_coarse,
+                rho=self.rho_coarse,
+                eps=self.eps_coarse,
+                reg_mode=self.reg_mode,
+                divergence=self.divergence,
+            )
+
+            fine_mapping = FUGWSparse(
+                alpha=self.alpha_fine,
+                rho=self.rho_fine,
+                eps=self.eps_fine,
+                reg_mode=self.reg_mode,
+                divergence=self.divergence,
+            )
+
+            coarse_to_fine.fit(
+                source_features=source_features_normalized,
+                target_features=target_features_normalized,
+                source_geometry_embeddings=geometry_embedding_normalized,
+                target_geometry_embeddings=geometry_embedding_normalized,
+                source_sample=sampled_geometry,
+                target_sample=sampled_geometry,
+                coarse_mapping=coarse_mapping,
+                source_selection_radius=(radius / max_distance),
+                target_selection_radius=(radius / max_distance),
+                fine_mapping=fine_mapping,
+                device=device,
+                verbose=verbose,
+                **kwargs,
+            )
+
+            self.mapping = fine_mapping
+
+        return self
+
+    def transform(
+        self,
+        X,
+        id_reg=0.0,
+        device="auto",
+    ):
+        """Project features using the fitted FUGW alignment
+
+        Parameters
+        ----------
+        X : ndarray of shape (n_samples, n_features)
+            Source features
+        id_reg: float, in the [0, 1] interval, defaults to 0
```

Comment:
these 2 parameters should be specified at the class creation step I think.

Reply:
I agree that currently, it aligns with the FUGW API, where initialization adheres to the BaseMapping class:
```
class BaseMapping:
    def __init__(
        self,
        alpha=0.5,
        rho=1,
        eps=1e-2,
        reg_mode="joint",
        divergence="kl",
    ):
```
 Adopting a more sklearn-like approach could be advantageous, yet I haven't explored it in detail.